src/sentry/workflow_engine/endpoints/organization_detector_index.py

```diff
@@ -509,7 +510,20 @@ def delete(self, request: Request, organization: Organization) -> Response:
         if not can_edit_detectors(queryset, request):
```

Comment:
I wonder if we should have a `can_delete_detectors` function for this? We should also lock down the `DELETE /detectors/<id>/` endpoint.

I think for this API it would be better if we rejected if any system-created detectors are included in the queryset. That way the results are more consistent (since we already reject if you don't have edit permissions for some of the detectors)